### PR TITLE
Patch and test for Issue #313.

### PR DESCRIPTION
**Pull request by traiansf**

### CODE REVIEW

File: src/javasources/KTool/src/org/kframework/compile/transformers/ResolveFunctions.java [hunk 1]
```diff
@@ -33,8 +37,27 @@ public ASTNode transform(Rule node) throws TransformerException {
         if (body instanceof KApp) {
             Term l = ((KApp) body).getLabel();
             if (l instanceof KLabelConstant) {
-                if (((KLabelConstant)l).isPredicate()) {
+                KLabelConstant label = (KLabelConstant) l;
+                if (label.isPredicate()) {
                     node = addFunction(node);
+                } else {
```
Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/compile/transformers/ResolveFunctions.java]: it looks like you're trying to mimic having some kind of label.isFunction method without actually having one. Is there a reason you didn't go that route?

Comment (traiansf) [inline src/javasources/KTool/src/org/kframework/compile/transformers/ResolveFunctions.java]: No particular reason.  Would you like me to add a method isFunctional to KLabelConstant?

Then, what would be the decision in case a KLabel can be both functional and unfunctional -- would we still always report error?  Perhaps that would be fine, too.

Comment (dwightguth) [inline src/javasources/KTool/src/org/kframework/compile/transformers/ResolveFunctions.java]: Well, think about the semantics. If I have a production that has a particular label of arity N and it's a function, then any time I rewrite something to that klabel it will evaluate the function. Which means you can never actually match the non-functional production that has that label, unless it has a different arity, which is just weird and we don't want to support that. So yeah, I think it makes sense to do that. But we can discuss it on Monday briefly if you want.

**Comment by traiansf** [comment]
I've added an isFunctional method.  
@dwightguth: Could you take another look?

**Comment by dwightguth** [comment]
Looks good just one minor thing: can you use -- instead of - in your config.xml? It will make my life easier merging the JCommander stuff. But otherwise, go ahead and merge once you do that.

**Comment by traiansf** [comment]
retest this please

**Comment by traiansf** [comment]
@dwightguth: i've changed the xml as required, checkstyle passes on my machine, still it fails on Jenkins. Any ideas why?

**Comment by grosu** [comment]
Dwight, you cannot use -- in an XML file, not even in a string.  This sucks, I know, but that's how it is.  We had this discussion before you returned, and I could not believe it.  Then I checked and indeed, no way to write -- in an XML file.  Maybe we should not be that concerned with all the corner cases of our K parsing of bubbles; others suck even worse.

**Comment by dwightguth** [comment]
You can use it in strings, but not comments. But the arguments in favor of
ever commenting out tests are specious at best. We had bugs for a while
because people commented out tests when they broke. Also, it's not the only
language that doesn't allow arbitrary strings in multiline comments. K
doesn't either.

So I really think that -- is the right move here because it cleans up some
ugliness in ktest that makes it harder to maintain. but I'm telling you
this way per our conversation this morning.

**Comment by dwightguth** [comment]
Traian: as I said via IM, you want the first line to be an xml declaration, not an element. Note the "?" in `<?xml ?>`

**Comment by traiansf** [comment]
@dwightguth: of course, you were right all along.  I just thought the ? are part of the regular expression.  silly of me.
As we were just waiting for this to pass, I'm merging this/
